go-controller/pkg/node/gateway_init.go

```diff
@@ -116,16 +128,9 @@ func (n *OvnNode) initGateway(subnets []*net.IPNet, nodeAnnotator kube.Annotator
 		gatewayNextHop := net.ParseIP(config.Gateway.NextHop)
 		gatewayIntf := config.Gateway.Interface
 		if gatewayNextHop == nil || gatewayIntf == "" {
-			// We need to get the interface details from the default gateway.
-			defaultGatewayIntf, defaultGatewayNextHop, err := getDefaultGatewayInterfaceDetails()
```

Comment:
You need to move the code above this as well; you've removed the ability to override the nexthop/interface via the config

Reply:
I don't follow. If `config.Gateway.NextHop` or `config.Gateway.Interface` are set: we set and use the `gatewayNextHop` and `gatewayIntf` to `initSharedGateway`, just like before. Am I missing something?

Reply:
ah, I missed the distinction between `gatewayIntf` and `defaultGatewayIntf`.

Still, the way things are written now, if someone overrides `config.Gateway.Interface`, then that will not affect `n.defaultGatewayIntf`, which seems wrong